js/app/interfaces/card.js

```diff
@@ -365,18 +365,25 @@ const Card = new Lang.Interface({
      *
      * Sets up a frame to show the model's thumbnail uri.
      */
-    set_thumbnail_frame_from_model: function (frame) {
+    set_thumbnail_frame_from_model: function (frame, background_size='cover') {
         frame.visible = false;
         if (!this.model.thumbnail_uri)
             return;
+        if (background_size === 'center') {
+            let im = new Gtk.Image({
```

Comment:
Well this is only meant for background images which are small icons. Yea you could misuse it and would look bad but that's true about a number of our modules